R/db-odbc-mssql.R

```diff
@@ -100,3 +100,45 @@
   )
   DBI::dbExecute(con, sql)
 }
+
+#' @export
+`db_compute.Microsoft SQL Server` <- function(con,
+                                              table,
+                                              sql,
+                                              temporary = TRUE,
+                                              unique_indexes = list(),
+                                              indexes = list(),
+                                              ...)
+{
+  if(!is.list(indexes))
+    indexes <- as.list(indexes)
+
+  if(!is.list(unique_indexes))
+    unique_indexes <- as.list(unique_indexes)
+
+  db_save_query(con, sql, table, temporary = TRUE)
+  db_create_indexes(con, table, unique_indexes, unique=TRUE)
+  db_create_indexes(con, table, indexes, unique=FALSE)
+
+  table
+}
+
+
+#' @export
+`db_save_query.Microsoft SQL Server` <- function(con,
+                                                 sql,
+                                                 name,
+                                                 temporary = TRUE,
+                                                 ...)
+{
+  # check that name has prefixed '##' if temporary
+  if(temporary && substr(name, 1, 1) != "#") {
+    table <- paste0("##", table)
+    message("MS SQL temp tables names require a single or double pound sign prefix. Renaming table to: ", table)
+  }
+
+  tt_sql <- build_sql("select * into ", ident_q(name), " from (", sql, ") ", ident_q(name), con=con)
```

Comment:
* I think `ident_q` should be `as.sql` now
* Can you please put spaces around the `=`